maven-plugins/bnd-maven-plugin/src/main/java/aQute/bnd/maven/plugin/AbstractBndMavenPlugin.java

```diff
@@ -259,12 +266,16 @@ public void execute() throws MojoExecutionException, MojoFailureException {
 			// 'test-wrapper-bundle'
 			if (classesDir.isDirectory()) {
 
-				// delete existing MANIFEST.MF from previous build
-				// target/classes/META-INF is output and should not be picked up
-				File existingMetaInf = new File(classesDir, "META-INF/MANIFEST.MF");
-				if (existingMetaInf.exists()) {
-					logger.info("Delete existing manifest from previous build: " + existingMetaInf);
-					IO.delete(existingMetaInf);
+				if (deleteExistingManifest) {
+					// delete existing MANIFEST.MF
+					// e.g. from previous build to avoid that this MANIFEST.MF
+					// becpmes input to the bnd process (which could lead to
+					// suprising results in some cases)
+					File existingMetaInf = new File(classesDir, "META-INF/MANIFEST.MF");
+					if (existingMetaInf != null && existingMetaInf.exists()) {
```

Comment:
I don't think you need to do these checks. Just call IO.delete. If getManifestPath returns null, we have big problems and should NPE.